web/pages/index.tsx

```diff
@@ -1,35 +1,191 @@
-import { FacebookAuthProvider, signInWithPopup } from "firebase/auth";
-import Head from "next/head";
+import {
+  FacebookAuthProvider,
+  GoogleAuthProvider,
+  signInWithPopup,
+  signOut,
+  User,
+} from "firebase/auth";
+import {
+  Button,
+  Flex,
+  Stack,
+  Text,
+  Divider,
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  Link,
+} from "@chakra-ui/react";
 import { auth } from "../utils/firebase/firebaseClient";
-import { trpc } from "../utils/trpc";
+import React from "react";
+//import { trpc } from "../utils/trpc";
 import { useAuth } from "../context/auth";
 
 export default function Home() {
-  async function handleLogin() {
+  const { user } = useAuth();
+
+  async function handleLoginFacebook() {
     const provider = new FacebookAuthProvider();
     const result = await signInWithPopup(auth, provider);
     const user = result.user;
   }
 
-  const { user } = useAuth();
-  console.log(user);
+  async function handleLoginGoogle() {
+    const provider = new GoogleAuthProvider();
+    const result = await signInWithPopup(auth, provider);
+    const user = result.user;
+  }
 
-  const listPosts = trpc.post.list.useQuery();
-  console.log(listPosts.data);
+  //const listPosts = trpc.post.list.useQuery();
+  //console.log(listPosts.data);
+
+  if (user !== null) {
+    return (
+      <Flex height="100vh">
+        <Flex width="100%" justifyContent="center" alignItems="center">
+          <Button
+            cursor={["default", "pointer"]}
+            bgColor="#D9D9D9"
+            onClick={() => {
+              signOut(auth);
+            }}
+          >
+            Logout
+          </Button>
+        </Flex>
+      </Flex>
+    );
+  }
 
   return (
-    <>
-      <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-        <meta name="viewport" content="width=device-width, initial-scale=1" />
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
-      <main>
-        <div>
-          <button onClick={handleLogin}>Login</button>
-        </div>
-      </main>
-    </>
+    <Flex height="100vh">
+      <Stack direction="row" width="100%" height="100%">
+        <Flex bgColor="#D9D9D9" width={["0%", "50%"]}></Flex>
+        <Flex bgColor="#FFFFFF" width={["100%", "50%"]} justifyContent="center">
+          <Stack direction="column" width="100%" alignItems="center">
+            <Stack
+              direction="row"
+              justifyContent="flex-end"
+              width="100%"
+              padding="2%"
+              marginBottom={["30%", "15%"]}
+            >
+              <Flex
+                bgColor="#D9D9D9"
+                position="absolute"
+                paddingRight="30px"
+                paddingLeft="30px"
+                paddingTop="10px"
+                paddingBottom="10px"
```

Comment:
Use the `top`, `left`, `right`, and `bottom` attributes to anchor the element instead of padding. Read [this](https://developer.mozilla.org/en-US/docs/Web/CSS/position) page for more info. See below for the other location to update as well.